tasks/html.js

```diff
@@ -6,42 +6,119 @@
  * Licensed under the MIT license.
  */
 
+var path = require('path');
+var chalk = require('chalk');
+var hooker = require('hooker');
 var htmllint = require('../lib/htmllint');
 
 module.exports = function(grunt) {
   "use strict";
 
-  var chalk = require('chalk');
+  var usingDefaultReporter,
+
+    // Default Grunt reporter
+    defaultReporter = function(result) {
+      result.forEach(function(message) {
+        var output = chalk.cyan(message.file) + ' ';
+        output += chalk.red('[') + chalk.yellow('L' + message.lastLine) +
+          chalk.red(':') + chalk.yellow('C' + message.lastColumn) + chalk.red('] ');
+        output += message.message;
+        grunt.log.writeln(output);
+      });
+    },
+
+    // Select a reporter (if not using the default Grunt reporter)
+    selectReporter = function(options) {
+      switch (true) {
+        // Checkstyle (XML) reporter
+        case options.reporter === 'checkstyle':
+          options.reporter = '../lib/reporters/checkstyle.js';
+          break;
+
+        // JSON reporter
+        case options.reporter === 'json':
+          options.reporter = '../lib/reporters/json.js';
+          break;
+
+        // Custom reporter
+        case options.reporter != null:
+          options.reporter = path.resolve(process.cwd(), options.reporter);
+      }
+
+      var reporter;
+      if (options.reporter) {
+        try {
+          reporter = require(options.reporter).reporter;
+          usingDefaultReporter = false;
+        } catch (err) {
+          grunt.fatal(err);
+        }
+      }
+
+      if (!reporter) {
+        reporter = defaultReporter;
+        usingDefaultReporter = true;
+      }
+
+      return reporter;
+    };
 
   grunt.registerMultiTask('htmllint', 'Validate html files', function() {
     var done = this.async(),
       files = grunt.file.expand(this.filesSrc),
       options = this.options({
         files: files,
-        force: false
+        force: false,
+        absoluteFilePathsForReporter: false
       }),
-      force = options.force;
+      force = options.force,
+      reporter = selectReporter(options),
+      reporterOutput = options.reporterOutput,
+      output;
+
+    // Hook into stdout to capture report
```

Comment:
That's just it, though: the only checkstyle formatters available through npm are intended for use with JSHint, so they all expect a different result object structure, including error codes which they prefix with `jshint.`, and they all write to `stdout`.

The best solution, I think, is to maintain our own formatters and have them simply return a String rather than writing to any output buffers. Then we can process that output in `html.js` as need be.